templates/search_supervisor.html

```diff
@@ -0,0 +1,56 @@
+<style>
+    .badge a {
+        color: black;
+        text-decoration: none;
+    }
+
+    .badge a:hover {
+        color: red;
+    }
+</style>
+
+<select class="form-select" id="supervisor" name="supervisor"
+        {% if page == 'profile' %}
```

Comment:
This is not a modular practice to check who called the template. This code is easily breakable by changing the page name. Besides, the more pages you call this template from, the more `if` statements you will have. You then have multiple-feature-related code blocks in a generic file. This can easily become dead-code if you remove those features without full analysis.

The best design is to make your template depend on arguments/parameters and then include this template using a `{% with %}` statement. 

You can also rely on inheritance if only arguments are not convenient. In the latter case, you define the common code with empty `{% block %}` statements, and define a feature/page-specific template inheriting from the common code, and implementing the specific `{% block %}` parts.

Here you seem to have two main paramters : 
- whether you need to disable the select
- whether researcher is defined and need to show currently assigned supervisors.

The code should reflect those ideas to remain clear and maintainable for years.